frontend/src/app/features/work-packages/components/wp-single-view/wp-single-view.component.ts

```diff
@@ -219,7 +214,28 @@ export class WorkPackageSingleViewComponent extends UntilDestroyedMixin implemen
         href: this.PathHelper.projectWorkPackagePath(project.id, workPackageId),
         matches: project.href === this.currentProject.apiv3Path,
       };
+    }
+
+    if (isNewResource(resource)) {
+      this.updateWorkPackageCreationState(change);
+    }
+
+    // eslint-disable-next-line no-underscore-dangle
+    this.groupedFields = this.rebuildGroupedFields(change, this.schema(resource)._attributeGroups) as GroupDescriptor[];
+    this.cdRef.detectChanges();
+  }
 
+  private updateWorkPackageCreationState(change:WorkPackageChangeset) {
```

Comment:
Should this be re-ran whenever the project changes? e.g. at `/work_packages/new?type=1`

Reply:
this is executed everytime anything in the resource is changed. I merely extracted the previous hooks (prev. on line 241 and 203) into another method.

But yes, the change you describe is the root cause, why this needs updating. But also, it must be executed in the beginning, e.g. when opening the wp creation in split view (`/projects/testproject/work_packages/create_new?type=1`). This is, why we can't simply listen to project property changes in the changeset.

I also extracted it to separate it more from the rest of the context, which is executed for any work package single view, e.g. in wp details page. This code, this update must only be run while creation of new work packages.

It does have a bad taste to it, but mostly related to the fact, that this component is used in so many different contexts.